fast-cart/src/components/Home.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import service from '../service'
+import {apiWrapper} from '../ulility-functions';
+import Loading from './Loading';
+import ProductGrid from './ProductGrid';
+import SideBar from './SideBar';
+
+const Home = () => {
+
+    const [productList, setProductList] = useState(undefined);
+
+    useEffect(()=>{
+        getAllProducts();
+    },[]);
+    const  getAllProducts = async () => {
+        let res =  await apiWrapper(service.getAllProducts,[]);
+        setProductList(res.products);
+    }
+
+    
+    return ( 
+        <div className='home'>
+            <SideBar/>
+            { productList === undefined &&
+                <Loading/>
+            }
+            { productList !== undefined &&
+                <ProductGrid productList={productList}/>
+            } 
+        </div>
+        
+    );
+}
+ 
+export default Home;
```

Comment:
I would suggest for handling scenarios where we make api calls and render data from that, follow this approach

always keep a state or include a key in existing state, where we have _init, loading, success and error_ condition,
so currently for some reason if your api gets failed, you will show infinite loader without informing user that error has occurred so he can refresh or something.

so your component can have these stages
init state => api call in progress => success or error state

so based on the requirement we can render different UIs